### PR TITLE
Add cccl recipe (rename from cuda-cccl-impl).

### PR DESCRIPTION
**Pull request by bdice**
This PR adds the `cccl` package, which is a rename of the `cuda-cccl-impl` package (identical contents for now). Due to some upstream reorganization of the CCCL libraries (thrust, cub, libcudacxx) it now makes sense to make `cccl` the proper name for this user-facing package, instead of `cuda-cccl-impl` which sounds like an internal "implementation" package.

This rename has been discussed with stakeholders/maintainers here: https://github.com/conda-forge/cuda-cccl-impl-feedstock/issues/2

This rename is being opened as a new staged recipe in accordance with this guide: https://conda-forge.org/docs/orga/guidelines.html#renaming-packages

For reference, the `cuda-cccl-impl` package was previously added in https://github.com/conda-forge/staged-recipes/pull/21953.

After this PR is merged, I will follow up with a few downstream recipes to update them.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/cccl/meta.yaml
Comment (jakirkham) [inline recipes/cccl/meta.yaml:39]:
```suggestion

requirements:
  run_constrained:
    # Prevent clobbering with `coda-cccl-impl`
    - cuda-cccl-impl <0a0

```

**Comment by bdice** [comment]
@adibbley  Can you comment here to indicate you're willing to be listed as a maintainer?

Also cc: @jakirkham @jrhemstad for visibility

**Review by jakirkham — COMMENTED** [review]
Thanks Bradley! 🙏

Think we will want to add `run_constrained` sections to both recipes to prevent clobbering

**Comment by jakirkham** [comment]
cc @adibbley

File: recipes/cccl/meta.yaml
Comment (bdice) [reply recipes/cccl/meta.yaml:39]: Resolved in 393c500.

**Comment by adibbley** [comment]
> @adibbley Can you comment here to indicate you're willing to be listed as a maintainer?

Yes, thanks!

File: recipes/cccl/meta.yaml
Comment (jakirkham) [inline recipes/cccl/meta.yaml:73]: Please feel free to add me here

```suggestion
    - bdice
    - jakirkham
```

**Comment by bdice** [comment]
@adibbley @jakirkham Anything else we need here before requesting a final review/merge?

**Review by adibbley — APPROVED** [review]

File: recipes/cccl/meta.yaml
Comment (jakirkham) [inline recipes/cccl/meta.yaml:63]: Thrust also uses [the Boost license]( https://github.com/NVIDIA/thrust/blob/f514806bbab3bb6439e63c64ac196b82319943f1/LICENSE#L195 ) and [FreeBSD or BSD-2-Clause license]( https://github.com/NVIDIA/thrust/blob/f514806bbab3bb6439e63c64ac196b82319943f1/LICENSE#L221 )

Also `libcudacxx` uses the [NCSA license]( https://github.com/NVIDIA/libcudacxx/blob/538a6a57bb155ea7a0a5d8371b7b5480b5a963f1/LICENSE.TXT#L249 ) and [MIT license]( https://github.com/NVIDIA/libcudacxx/blob/538a6a57bb155ea7a0a5d8371b7b5480b5a963f1/LICENSE.TXT#L241 ). Finally there is an [NVIDIA license]( https://github.com/NVIDIA/libcudacxx/blob/538a6a57bb155ea7a0a5d8371b7b5480b5a963f1/LICENSE.TXT#L320 ).

```suggestion
  license: Apache-2.0 AND BSD-3-Clause AND BSD-2-Clause AND BSL-1.0 AND NCSA AND MIT AND LicenseRef-NVIDIA-Software-License
```

File: recipes/cccl/meta.yaml
Comment (jakirkham) [inline recipes/cccl/meta.yaml:67]: Maybe we can stick to the `summary` and skip the `description` since they are the same

```suggestion
  summary: CUDA C++ Core Libraries
```

File: recipes/cccl/build.sh
Comment (jakirkham) [inline recipes/cccl/build.sh:13]: Wonder if there is a CMake install approach we can use here in the future

**Review by jakirkham — COMMENTED** [review]
Thanks Bradley! 🙏

Had a few suggestions below

File: recipes/cccl/build.sh
Comment (bdice) [reply recipes/cccl/build.sh:13]: This can be reworked with CCCL 2.2 which releases from the new monorepo. Let’s keep the status quo for now since this is just renaming a package we have already tested/used a bit and trust.

**Comment by jakirkham** [comment]
Thanks Bradley & Alex! 🙏

**Pull request merged**
